napari/layers/_layer_actions.py

```diff
@@ -239,3 +239,40 @@ def _project(ll: LayerList, axis: int = 0, mode: str = 'max') -> None:
     )
 
     ll.append(new)
+
+
+def _enable_measure_shapes(ll: LayerList):
+    def polygon_area(pts):
+        x, y = pts.T
+        correction = x[-1] * y[0] - y[-1] * x[0]
+        main_area = np.dot(x[:-1], y[1:]) - np.dot(y[:-1], x[1:])
+        return 0.5 * np.abs(main_area + correction)
+
+    def measure(s_type, data):
```

Comment:
As I understand, the function is returning length of edges and shape area?

Reply:
Yes, perimeter and area.